website/content/docs/platform/k8s/vso/auth/gcp.mdx

```diff
@@ -0,0 +1,67 @@
+---
+layout: docs
+page_title: Vault Secrets Operator GCP Auth
+description: >-
+  The Vault Secrets Operator supports authenticating to Vault with the GCP auth method
+---
+
+# GCP auth method support
+
+The Vault Secrets Operator (VSO) supports authenticating to Vault's GCP auth method, using GKE workload identity.
+
+See Google's [Use Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) for the steps necessary to enable workload identity on a GKE cluster, and to allow a Kubernetes service account to impersonate a Google IAM service account.
+
+Once workload identity is configured on a GKE cluster, use a Vault role config like this:
+
+<CodeTabs>
+<CodeBlockConfig>
+
+```shell-session
+$ vault write auth/gcp/role/<VAULT_GCP_ROLE> \
+    type="iam" \
+    policies="default" \
+    max_jwt_exp=3600 \
+    bound_service_accounts="<SERVICE_ACCOUNT>@<GCP_PROJECT>.iam.gserviceaccount.com"
+```
+
+</CodeBlockConfig>
+<CodeBlockConfig>
+
+```hcl
+resource "vault_gcp_auth_backend_role" "gcp_role" {
+  backend   = "auth/gcp"
+  role      = <VAULT_GCP_ROLE>
+  type      = "iam"
+  token_policies = "default"
+  max_jwt_exp = 3600
+  bound_service_accounts = [
+    "<SERVICE_ACCOUNT>@<GCP_PROJECT>.iam.gserviceaccount.com",
+  ]
+}
+```
+
+</CodeBlockConfig>
+</CodeTabs>
+
+<Note>
+`max_jwt_exp` needs to be greater than or equal to 1 hour (3600)
+</Note>
+
+Then use the auth role with a VaultAuth like this:
+
+```yaml
+apiVersion: secrets.hashicorp.com/v1beta1
+kind: VaultAuth
+metadata:
+  name: vaultauth-gcp-example
+  namespace: <K8S_NAMESPACE>
+spec:
+  vaultConnectionRef: <VAULT_CONNECTION_NAME>
+  mount: gcp
+  method: gcp
+  gcp:
+    role: <VAULT_GCP_ROLE>
+    workloadIdentityServiceAccount: <SERVICE_ACCOUNT>
+```
+
+See the full list of gcp auth options on the [api page](https://github.com/hashicorp/vault-secrets-operator/blob/c9eb11f776c7f86895383fc32740bbb73e141aa5/docs/api/api-reference.md#vaultauthconfiggcp).
```

Comment:
Should this link to the API reference docs instead? I don't see them added to this PR yet though, so perhaps I'm missing something.

Reply:
Oh yep, for sure. The API docs will go out as part of the operator release, so we can add the real link then.